### PR TITLE
[LDR] Introduce a private heap for the loader.

### PR DESCRIPTION
**Pull request by learn-more**

### CODE REVIEW

File: dll/ntdll/ldr/ldrinit.c [hunk 1]
```diff
@@ -1875,8 +1876,15 @@ LdrpInitializeProcess(IN PCONTEXT Context,
     Status = RtlAllocateActivationContextStack(&Teb->ActivationContextStackPointer);
     if (!NT_SUCCESS(Status)) return Status;
 
-    // FIXME: Loader private heap is missing
-    //DPRINT1("Loader private heap is missing\n");
+    RtlZeroMemory(&HeapParameters, sizeof(HeapParameters));
+    HeapFlags = HEAP_GROWABLE | HEAP_CLASS_1;
+    HeapParameters.Length = sizeof(RTL_HEAP_PARAMETERS);
+    LdrpHeap = RtlCreateHeap(HeapFlags, 0, 0x10000, 6000, 0, &HeapParameters);
```
Comment (ThFabba) [inline dll/ntdll/ldr/ldrinit.c]: 6000? Shouldn't CommitSize be divisible by PAGE_SIZE?

File: dll/ntdll/ldr/ldrinit.c [hunk 2]
```diff
@@ -1875,8 +1876,15 @@ LdrpInitializeProcess(IN PCONTEXT Context,
     Status = RtlAllocateActivationContextStack(&Teb->ActivationContextStackPointer);
     if (!NT_SUCCESS(Status)) return Status;
 
-    // FIXME: Loader private heap is missing
-    //DPRINT1("Loader private heap is missing\n");
+    RtlZeroMemory(&HeapParameters, sizeof(HeapParameters));
+    HeapFlags = HEAP_GROWABLE | HEAP_CLASS_1;
+    HeapParameters.Length = sizeof(RTL_HEAP_PARAMETERS);
```
Comment (ThFabba) [inline dll/ntdll/ldr/ldrinit.c]: Why not `sizeof(HeapParameters)` here like above?

File: dll/ntdll/ldr/ldrinit.c (hunk 1 above)
Comment (HBelusca) [reply dll/ntdll/ldr/ldrinit.c]: Can we use some #define'd values there?

**Comment by HBelusca** [comment]
Can you add to the commit message the reason why it is interesting to use such a private heap?

**Pull request merged**